tools/score_instrument.php

```diff
@@ -51,74 +68,67 @@
-$client->makeCommandLine();
-$client->initialize("../project/config.xml");
-
-$today= getdate();
-$date = sprintf("%04d-%02d-%02d", $today['year'], $today['mon'], $today['mday']);
-
-$logfp = fopen("logs/score_instrument.$date.log", 'a');
-function log_msg($message) {
-    global $logfp;
-    $now_array = getdate();
-    $now_string = sprintf("%04d-%02d-%02d %02d:%02d:%02d", $now_array['year'], $now_array['mon'], $now_array['mday'], $now_array['hours'], $now_array['minutes'], $now_array['seconds']);
-    fwrite($logfp, "[$now_string] $message\n");
-}
-
 // include instrument class
 if(strtolower($test_name) != 'all') {
-     if (!is_file("../project/instruments/NDB_BVL_Instrument_$test_name.class.inc")
-            && !is_file("../project/instruments/$test_name.linst")) {
-        fwrite(STDERR, "Included file does not exist (../project/instruments/NDB_BVL_Instrument_$test_name.class.inc)\n");
+    if (!is_file("../project/instruments/NDB_BVL_Instrument_$test_name.class.inc")
+        && !is_file("../project/instruments/$test_name.linst")) {
+        printError("Included file does not exist (../project/instruments/NDB_BVL_Instrument_$test_name.class.inc)");
         return false;
     }
 }
 
-$db =& Database::singleton();
-
 // check that the $test_name is a valid instrument
 if($test_name== 'all') {
     $query = "SELECT DISTINCT test_name FROM test_battery WHERE Active='Y'";
 } else {
     $query = "SELECT DISTINCT test_name FROM test_battery WHERE Active='Y' AND Test_name = :tnm";
 }
-$result = $db->pselect($query, array('tnm' => $test_name));
+$result = $DB->pselect($query, array('tnm' => $test_name));
+
 // if nothing is returned than the instrument DNE
 if (!is_array($result) || count($result)==0) {
-    fwrite(STDERR, "Invalid Instrument ($test_name)!\n");
+    printError("Invalid Instrument ($test_name)!");
     return false;
 }
 
-
 // get the list of CommentIDs for valid timepoints
 foreach($result as $test) {
     $test_name = $test['test_name'];
-    fwrite(STDERR, "Running scoring for $test_name");
-    log_msg("");
-    log_msg("Running scoring for $test_name");
-    log_msg("------------------------------");
-
-    $query = "SELECT s.CandID, s.Visit_label, s.ID as SessionID, t.CommentID, c.PSCID
-        FROM candidate as c, session as s, flag as f, $test_name as t
-        WHERE c.CandID=s.CandID AND s.ID=f.SessionID AND f.CommentID=t.CommentID
-        AND s.Active = 'Y' AND c.Active='Y' 
-        AND f.Test_name = :tnm AND f.Administration <> 'None' AND f.Administration IS NOT NULL";
-
+    printOut("Running scoring for $test_name");
+    logMessage("");
+    logMessage("Running scoring for $test_name");
+    logMessage("------------------------------");
+
+    $query = "
+        SELECT s.CandID, 
+            s.Visit_label, 
+            s.ID as SessionID, 
+            t.CommentID, 
+            c.PSCID
+        FROM candidate as c, 
+            session as s, 
+            flag as f, 
+            $test_name as t
```

Comment:
should not use a cross-join but an inner joint

FROM candidate c,
JOIN session s ON c.CandID = s,CandID
JOIN flag f ON s.ID = f.SessionID 
...

If some table might not have actual data, you can use outer join : LEFT JOIN table ON .... or RIGHT JOIN table ON ...

Reply:
that might chnage functionality and would then need to be sent to minor, the idea here is to maintain current functionality and just make it runnable

Reply:
It's already changing the functionality, making them LEFT JOINs with an ON clause is equivalent to cross joins with a WHERE clause (in fact, the WHERE clause is more likely to accidentally filter things out.)

Reply:
Inner join are more effective than cross joint with the same final result